### PR TITLE
Renamed impl files to private and updated code style document

### LINKED ISSUE #1711: Rename *_impl.* to *_private.* and add that to the code conventions
### Description of the task
*_impl.* files are used for private functions for a class/file that should not be exposed to users
impl can be confused with [pimpl](https://en.cppreference.com/w/cpp/language/pimpl), so using private in the naming is more clear

### Acceptance criteria

- [ ] Rename *_impl.* to *_private.*
- [ ] Document this in code conventions

### PR DESCRIPTION
**Pull request by HubertTheodore**
### Description

    renamed _impl files to _private and updated code style guide document to reflect changes

### Resolved Issues

resolves #1711 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ x] **Remove all commented out code**
- [ x] **Remove extra print statements**: for example, those just used for testing
- [ x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
Something funky is going on with the Software submodule, try to delete the extra empty Software folder in the repo and make a new commit

You also need to update src/firmware/app/control/BUILD with the new file name

File: src/firmware/app/control/trajectory_planner_private.c [hunk 1]
```diff
@@ -1,10 +1,9 @@
-#include "firmware/app/control/trajectory_planner_impl.h"
-
 #include <assert.h>
 #include <float.h>
 #include <math.h>
 
 #include "firmware/app/control/trajectory_planner.h"
+#include "firmware/app/control/trajectory_planner_impl.h"
```
Comment (LiCody) [inline src/firmware/app/control/trajectory_planner_private.c]:
```suggestion
#include "firmware/app/control/trajectory_planner_private.h"
```
Build is failing cause it cant find the _impl files should change these to use the renamed files

File: src/firmware/app/control/trajectory_planner_private_test.cpp [hunk 2]
```diff
@@ -1,8 +1,7 @@
 extern "C"
 {
-#include "firmware/app/control/trajectory_planner_impl.h"
-
 #include "firmware/app/control/trajectory_planner.h"
+#include "firmware/app/control/trajectory_planner_impl.h"
```
Comment (LiCody) [inline src/firmware/app/control/trajectory_planner_private_test.cpp]:
```suggestion
#include "firmware/app/control/trajectory_planner_private.h"
```
same as above

**Review by LiCody — CHANGES_REQUESTED** [review]
Looks good just need these changes to fix CI. Generally, you should run bazel build on your own machine before pushing a commit/opening a pr.

**Review by LiCody — APPROVED** [review]
LGTM 👍

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**